### PR TITLE
Issue #3261513 by nkoporec: Error: Call to undefined method Drupal\data_policy\DataPolicyConsentManager::hasGivenConsent()

### PR DESCRIPTION
**Pull request by nkoporec**
## Problem
When trying to login with social_magic_link, certain users receive this error:
`Error: Call to undefined method Drupal\data_policy\DataPolicyConsentManager::hasGivenConsent()`

## Solution
The hasGivenConsent() method no longer exists in data_policy, so we need to update it.

## Issue tracker
https://www.drupal.org/project/social/issues/3261513

## How to test
1. Create a new user, but don't login yet
2. Make sure data_policy module is enabled
2. Create a magic login link using social_magic_link module
3. Visit the magic link

## Screenshots
*If this Pull Request makes visual changes then please include some screenshots that show what has changed here. A before and after screenshot helps the reviewer determine what changes were made.*

## Release notes
*A short summary of the changes that were made that can be included in release notes.*

## Change Record
*If this Pull Request changes the way that developers should do things or introduces a new API for developers then a change record to document this is needed. Please provide a draft for a change record or a link to an unpublished change record below. [Existing change records](https://www.drupal.org/list-changes/social) can be consulted as example.*

## Translations
*Translatable strings are always extracted from the latest development branch. To ensure translations remain available for platforms running older versions of Open Social the original string should be added to `translations.php` when it's changed or removed.*
- [ ] Changed or removed source strings are added to the `translations.php` file.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2728-8bnf2cgielfvdn5mfrnbdrl6utfpcrn5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f7c2a200c7e8a758e4925b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2728-8bnf2cgielfvdn5mfrnbdrl6utfpcrn5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f7c2a200c7e8a758e4925b

File: modules/custom/social_magic_login/src/Controller/MagicLoginController.php
Comment (NickDaelemans) [inline modules/custom/social_magic_login/src/Controller/MagicLoginController.php:194]: Looks like a typo here.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2728-8bnf2cgielfvdn5mfrnbdrl6utfpcrn5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f7c2a200c7e8a758e4925b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2728-8bnf2cgielfvdn5mfrnbdrl6utfpcrn5.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61f7c2a200c7e8a758e4925b

**Review by NickDaelemans — APPROVED** [review]

**Comment by Ressinel** [comment]
Hi, @NickDaelemans 

Do not merge PRs that have errors in checks. Could you confirm with the Nucleus team before merging PR with check error? The error that this PR had is known, but it can be another one that can stop work for other devs by adding errors for their PRs.

Thanks!

**Pull request merged**
